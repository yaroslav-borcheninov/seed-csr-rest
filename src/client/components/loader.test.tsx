import React from "react"
import { shallow, ShallowWrapper } from "enzyme"

import Loader from "./loader"

jest.useFakeTimers()

describe("Loader", () => {
  const getComponent = () => shallow<Loader>(<Loader />)
  let component: ShallowWrapper<{}, {}, Loader>

  describe("when mounted", () => {
    beforeEach(() => {
      component = getComponent()
    })

    it("should render nothing", () => {
      expect(component).toMatchSnapshot()
    })

    it("should call setTimeout", () => {
      expect(window.setTimeout).toHaveBeenCalledWith(
        component.instance().showLoader,
        1000
      )
    })
  })

  describe("after 999ms", () => {
    beforeEach(() => {
      component = getComponent()
      jest.advanceTimersByTime(999)
    })

    it("should render nothing", () => {
      expect(component).toMatchSnapshot()
    })
  })

  describe("after 1001ms", () => {
    beforeEach(() => {
      component = getComponent()
      jest.advanceTimersByTime(1001)
    })

    it("should render", () => {
      expect(component).toMatchSnapshot()
    })
  })

  describe("when unmounted", () => {
    let instance: Loader

    beforeEach(() => {
      component = getComponent()
      instance = component.instance()
      component.unmount()
    })

    it("should call clearTimeout", () => {
      expect(window.clearTimeout).toHaveBeenCalledWith(instance.timeoutId)
    })
  })
})
