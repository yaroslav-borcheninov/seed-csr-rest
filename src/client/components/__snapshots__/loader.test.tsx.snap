// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Loader after 999ms should render nothing 1`] = `""`;

exports[`Loader after 1001ms should render 1`] = `
<div>
  Loading...
</div>
`;

exports[`Loader when mounted should render nothing 1`] = `""`;
